chain/stmgr/stmgr.go

```diff
@@ -1171,10 +1171,20 @@ func (sm *StateManager) GetFilVested(ctx context.Context, height abi.ChainEpoch,
 		}
 	}
 
-	// continue to use preIgnitionGenInfos, nothing changed at the Ignition epoch
-	vf = big.Add(vf, sm.preIgnitionGenInfos.genesisPledge)
-	// continue to use preIgnitionGenInfos, nothing changed at the Ignition epoch
-	vf = big.Add(vf, sm.preIgnitionGenInfos.genesisMarketFunds)
+	if height <= build.UpgradeActorsV2Height {
+		// continue to use preIgnitionGenInfos, nothing changed at the Ignition epoch
+		vf = big.Add(vf, sm.preIgnitionGenInfos.genesisPledge)
+		// continue to use preIgnitionGenInfos, nothing changed at the Ignition epoch
+		vf = big.Add(vf, sm.preIgnitionGenInfos.genesisMarketFunds)
+	} else {
+		ract, err := st.GetActor(builtin.ReserveAddress)
+		if err != nil {
+			return big.Zero(), xerrors.Errorf("failed to get reserve actor: %w", err)
+		}
+
+		// If money enters the reserve actor, this could lead to a negative term
+		vf = big.Add(vf, big.Sub(big.NewFromGo(build.InitialFilReserved), ract.Balance))
```

Comment:
Let's call this `ReserveDisbursed` as a separate term.

Reply:
Done